User Profile Page #24

bolt1003 opened the pull request:
Pushing the changes made to the user profile page to master. Database queries and tests still need to be completed. CSS, Routes, and Controller are mostly complete.

Commit:
push branch

Commit:
updated
controller

Commit:
updated
controller

Commit:
updated
controller

Commit:
updated routes

Commit:
updated routes

Commit:
updated routes

Commit:
added controller

Commit:
Pulled lastest changes

Commit:
Add comments

Commit:
Add Template URL

Commit:
Updated Routes & Controller

Commit:
Fixed 404 with username in URL

Commit:
Fleshed out public functions in the Profile Controller

Commit:
Created Database Migration

Commit:
Add Profile Page

Commit:
Redesigned the CSS on the user profile page

brec9824 commented on app/Http/Controllers/ProfileController.php:
```diff
@@ -0,0 +1,67 @@
+
+class ProfileController extends Controller
+{
+    public function profileView($username)
```
Since you have a user_profile db table setup. You should set up an [eloquent](https://laravel.com/docs/5.2/eloquent) relationship between user and user_profile. This will allow you to just pass a user object and you can access user_profile from it. No need to send username

rickbau5 commented on resources/views/profile/comments.blade.php:
```diff
@@ -0,0 +1,13 @@
+
+@section('mainBody')
+    <div>
+	<p> Testing </p>
```
@brec9824 has a better method, see below

brec9824 replied:
If you pass in a user object you can do

``` html
@foreach($user->comments as $comments)
            <h5>{{ $comment->comment_body }}</h5>
            <p>
                <br>
                @if ($comment->created_at == $comment->updated_at)
                    Created: {{ $comment->getCreatedAt() }}
                @else
                    Updated: {{ $comment->getUpdatedAt() }}
                @endif
            </p>
@endforeach
```

rickbau5 commented on app/Http/routes.php:
```diff
@@ -75,8 +75,24 @@
- 
- 
+
+Route::get('profile/view/{username}'		, 'ProfileController@profileView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
```
This validation shouldn't be done here. Usernames with illegal characters won't be created, and will 404 and that's totally fine.

rickbau5 commented on resources/views/profile/view.blade.php:
```diff
@@ -0,0 +1,46 @@
+@extends('layouts.main_layout')
```
formatting is all wonky in here. Try to get tags aligned properly.

rickbau5 commented on resources/views/profile/view.blade.php:
```diff
@@ -0,0 +1,46 @@
+				<div/>
+			<div/>
+		<div/>
+        <!-- Template: http://wrapbootstrap.com/preview/WB09JXK43 -->
```
also get rid of all of these comments...

brec9824 commented on app/Http/routes.php:
```diff
@@ -75,8 +75,24 @@
+Route::get('profile/comments/'			, 'ProfileController@commentsViewDefault');								// Implemented	
+Route::get('profile/{username}'			, 'ProfileController@profileView')		->where('username', '([a-zA-Z]{1,})([0-9]+)?\w+');	// Implemented
+Route::get('profile/'				, 'ProfileController@profileViewDefault');								// Implemented
+
```
If you want to check data and limit it do that in a controller or in a custom request.

rickbau5 commented:
Both this and #128 contain no functionality and therefore do not meet the requirement of this sprint as described [here](http://www2.cs.uidaho.edu/~jeffery/courses/383/hw7.html):
>Achieve (simple versions of) functionality for "many" (or "most") use cases. Whatever is essential for your vision.
I wouldn't consider a profile page that gives nothing more than a static page a "simple version of functionality". It's very close to being able to, as with #128 but the effort needs to be put in to get it there.

brec9824 commented on resources/views/profile/view.blade.php:
```diff
@@ -0,0 +1,46 @@
+		    	<div class="">
+					<h2 class="center"> {{$username}}</h2>
+					<div class="btn-group btn-group-justified">
+			        	<a href="#" class="btn btn-default">Send Message</a>
```
You should have a check here to see if the user is logged in and if they are the owner of this project. If they are you should hide this button.

brec9824 commented on resources/views/profile/view.blade.php:
```diff
@@ -0,0 +1,46 @@
+					<h2 class="center"> {{$username}}</h2>
+					<div class="btn-group btn-group-justified">
+			        	<a href="#" class="btn btn-default">Send Message</a>
+			        	<a href="#" class="btn btn-default">Add To Contact</a>
```
Same as above hide if the user shouldnt be seeing this button.

brec9824 commented on resources/views/profile/view.blade.php:
```diff
@@ -0,0 +1,46 @@
+		    	<div class="col-md-6">
+		        	<h4>Basic Information</h4>
+		            <label>User name:</label><p>{{$username}}</p><br/>
+		            <label>Birth Date:</label><p>{{$username }}</p><br/>
```
If you set up a eloquent relationship between user and user_profile. You can write something like this to get the data:
`html
<label>Birth Date:</label><p>{{ $user->profile->birthdate }}</p><br/>
`

This would apply to all the data in user_profile

brec9824 commented:
Look into eloquent(https://laravel.com/docs/5.2/eloquent) to set up the relationships to be able access these variables easily. Also you can look at ProjectComment.php, User.php, and Project.php they are all currently using eloquent relationships.

bolt1003 commented:
Thanks Brec, I will look into eloquent.

Commit:
Updated Migration, View Page, Controller, Routes

Commit:
Add Edit Form

Commit:
Updated Routes, Controller, Create user profile row on user creation

Commit:
Controller Revision, Edit Default Values, Routes added

Commit:
Edit Page Controller Modification

Commit:
Added User Avatar Directory

Commit:
Restore Git Ignore

Commit:
Updated User Profile Appearance

Commit:
Add user Profile Photo

Commit:
Minor Visual Modification

bolt1003 commented:
Thanks for your suggestions, the errors have been corrected. The profile page now pulls data from the database, modifies data in the profile table, and the theme is complete. The buttons "Send message" and "add to contacts" are now hidden when the user visits their own page and the edit function is hidden when a third party visits the user's page. Once the projects tab is complete the profile should be ready to be merged with master.

Commit:
Fixed Bug in Routes

brec9824 commented on public/css/custom.css:
```diff
@@ -83,6 +83,27 @@ Project list
     padding-right: 15px;
 }
 
+<<<<<<< HEAD
```
This needs to be removed

brec9824 commented on public/css/custom.css:
```diff
@@ -83,6 +83,27 @@ Project list
+.p-col-2-3 a{
+    padding-left: 20px;
+}
+=======
```
This needs to be removed. Its left over from rebase

brec9824 commented on resources/views/profile/comments.blade.php:
```diff
@@ -0,0 +1,11 @@
+
+@section('mainBody')
+    <div>
+	<p> Comments </p>
```
You can add all a users comments by doing:
[usercomments.txt](https://github.com/uidaho/squireproject/files/249082/usercomments.txt)

You just need to have access to the user of the comments you want to show and have the user object be $user

brec9824 commented on resources/views/profile/edit.blade.php:
```diff
@@ -0,0 +1,96 @@
+			</div>
+		</div>
+	</div>
+@stop
```
Formatting is off run Reformat code in phpstorm(Alt+F8)

brec9824 commented on database/migrations/2016_04_23_182146_create_user_profile_table.php:
```diff
@@ -0,0 +1,39 @@
+    {
+        Schema::drop('profiles');
+    }
+}
```
Formatting is off run Reformat code in phpstorm(Alt+F8)

brec9824 commented on app/Profile.php:
```diff
@@ -0,0 +1,22 @@
+        return '/images/users/' . $this->picture;
+    }
+
+}
```
Formatting is off run Reformat code in phpstorm(Alt+F8)

brec9824 commented on app/Http/Controllers/ProfileController.php:
```diff
@@ -3,39 +3,105 @@
+        return redirect('/profile/'.$user->username);
+    }
+
+}
```
Formatting is off run Reformat code in phpstorm(Alt+F8)

brec9824 commented on app/Http/Controllers/Auth/AuthController.php:
```diff
@@ -65,10 +67,19 @@ protected function validator(array $data)
+
+		return $user;	
     }
 }
```
Formatting is off run Reformat code in phpstorm(Alt+F8)

brec9824 commented on resources/views/profile/projects.blade.php:
```diff
@@ -0,0 +1,26 @@
+			<div class="panel-body">
+
+				Add Project List Here
+
```
You can use `@foreach($user->projects as $project)` to loop through all projects. You just need access to the user you want to display the projects for and have it in $user

brec9824 replied:
Done

brec9824 commented on resources/views/profile/view.blade.php:
```diff
@@ -1,24 +1,69 @@
+			</div>
+		</div>
+	</div>
+@stop
```
Formatting is off run Reformat code in phpstorm(Alt+F8)

brec9824 commented:
@bolt1003 Just some minor issues, but if you just get the files reformatted and the rebase code removed I would be willing to merge this. So that way we will have the user profile to show off today.

Some extra things (I wouldn't try to fix these because it will take to long):
- Edited profile data needs to be verified as valid and meeting requirements (So a new Request handler)
- Can manually put in edit profile route of another user to see their edit page (So need some auth checking)
- I dont see a way to edit profile picture
- Profile button on top right drop down doesnt work

Commit:
Added comments and projects

Added comments and projects to the profile pages. Also formatted some of unformatted code.

Commit:
Small change to navbar on profile

brec9824 commented:
Merging after this passes
